src/Source/DoctrineODMQuerySourceIterator.php

```diff
@@ -64,7 +75,30 @@ public function __construct(Query $query, array $fields, $dateTimeFormat = 'r')
             }
         }
 
-        $this->dateTimeFormat = $dateTimeFormat;
+        if (func_num_args() == 3 && is_string($dateTimeFormat)) {
+            // When removed, the code below initializing a default set of formatters should be rewritten as well
+            @trigger_error('Passing a dateTimeFormat string is deprecated since 1.7.2, to be removed in 2.0. '.
+                'Pass an array of DataFormatterInterfaces instead',
+                E_USER_DEPRECATED);
+        }
+
+        $formatters = array(new ArrayFormatter(), new DateTimeFormatter($dateTimeFormat), new ObjectToStringFormatter());
+        if (func_num_args() == 4) {
+            $additionalFormatters = func_get_arg(4);
+            if(!is_array($additionalFormatters)) {
+                throw new InvalidArgumentException('4th parameter must be an array of ' . DataFormatterInterface::class . ' classes');
+            }
+
+            $formatters = array_merge($formatters, $additionalFormatters);
+        }
+
+        foreach($formatters as $formatter) {
+            if(!$formatter instanceof DataFormatterInterface) {
```

Comment:
I don't think it needs a priority queue - Mainly because this really should be a very simple ordered array, and when using xdebug, a priority queue is opaque so for anyone debugging for whatever reason its a huge pain.